### DISCUSSION TITLE
which class contains the logic for handling environment variables

### DISCUSSION THREAD

**Discussion by mitht** [open-telemetry/opentelemetry-java-instrumentation#11832] [Q&A] [answered]
I want to directly embed some custom configurations into the source code of opentelemetry-java-instrumentation. I couldn't find any code handling environment variables in the opentelemetry-java-instrumentation/instrumentation/resources package. Could you please inform me where the code for handling environment variables is located?

The configuration added in the io.opentelemetry.instrumentation.resources.ProcessResource class does not take effect.

<img width="1529" alt="image" src="https://github.com/user-attachments/assets/8cb2cc69-9cc5-4f4f-89c4-3446b7344d35">

**Comment by laurit** [comment, marked as answer]
`RuntimeMXBean.getSystemProperties()` copies the system properties to a new map, modifying it won't help at all. The conventional way for programmatically setting a system property would be to use https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#setProperty-java.lang.String-java.lang.String- Though you shouldn't use that either. See https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/b1e4c32ac6d9c08d8e0d537a6cd33b71174cb65c/examples/extension/src/main/java/com/example/javaagent/DemoAutoConfigurationCustomizerProvider.java#L49-L56
